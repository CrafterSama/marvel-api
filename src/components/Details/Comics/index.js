import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { SpinnerCircularFixed } from 'spinners-react';
import { getComicById } from '../../../context/actions';
import { useStateValue } from '../../../context';
import { LoadingBox, Container } from '../../../styles';
import ComicDetails from './components/ComicDetails';

const Comics = () => {

  const { id, } = useParams();
  const [state, dispatch] = useStateValue();

  useEffect(() => {
    if (!state.comic) {
      getComicById(id, dispatch);
    }
    // eslint-disable-next-line
  }, [id]);

  const isEmpty = Object.entries(state.comic).length === 0;

  if (isEmpty) {
    return (
      <Container>
        <LoadingBox>
          <SpinnerCircularFixed
            size={100}
            thickness={100}
            speed={100}
            color='rgba(236, 29, 36, 1)'
            secondaryColor='rgba(204, 204, 204, 1)'
          />
        </LoadingBox>
      </Container>
    );
  };

  return (
    <Container>
      {state.comic && state.comic.map((comic, key) => (
        <ComicDetails data={comic} key={key} />
      ))}
    </Container>
  );

};

export default Comics;
