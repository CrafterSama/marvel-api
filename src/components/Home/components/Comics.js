import React from 'react'
import { useParams } from 'react-router-dom'
import ComicsDetails from './ComicDetails'
import { useStateValue } from '../../../context'


const Comics = () => {

  const { id } = useParams()
  const [state] = useStateValue()

  console.log(id, state)

  return <div />
  //return <ComicsDetails data={state.comics} />

}

export default Comics
