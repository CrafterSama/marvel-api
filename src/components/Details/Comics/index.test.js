import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Comics from './index';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({ id: 77117 }),
}));

describe('Test for <Comics /> Component', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <Comics />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
