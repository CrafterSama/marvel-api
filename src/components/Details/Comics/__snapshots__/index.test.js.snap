// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <Comics /> Component matches snapshot 1`] = `
<styles__Container>
  <styles__LoadingBox>
    <Wrapper
      color="rgba(236, 29, 36, 1)"
      enabled={true}
      secondaryColor="rgba(204, 204, 204, 1)"
      size={100}
      speed={100}
      style={Object {}}
      thickness={100}
    />
  </styles__LoadingBox>
</styles__Container>
`;
